op-batcher/batcher/driver.go

```diff
@@ -334,22 +334,37 @@ func (l *BatchSubmitter) publishTxToL1(ctx context.Context, queue *txmgr.Queue[t
 	return nil
 }
 
-// sendTransaction creates & submits a transaction to the batch inbox address with the given `data`.
+// sendTransaction creates & submits a transaction to the batch inbox address with the given `txData`.
 // It currently uses the underlying `txmgr` to handle transaction sending & price management.
 // This is a blocking method. It should not be called concurrently.
 func (l *BatchSubmitter) sendTransaction(txdata txData, queue *txmgr.Queue[txData], receiptsCh chan txmgr.TxReceipt[txData]) {
 	// Do the gas estimation offline. A value of 0 will cause the [txmgr] to estimate the gas limit.
 	data := txdata.Bytes()
+
+	var blobs []*eth.Blob
```

Comment:
Why is this safe to operate on a local timestamp unlike other hardfork logic that operates on a L1 based timestamp?

Reply:
Good point. This triggers whether to send a blob instead of regular transaction for the batch. If the L1 state is actually behind our local time then we could end up with a blob transaction included in a pre-Blob activation block that would get ignored by chain derivation leading to l2 reorg.

We could explicitly check to make sure chain tip is beyond BlobsEnabledL1Timestamp, though I guess to be super certain we won't have an L2 re-org because of this, we should wait some # of confirmations after?

Reply:
Note:  In a subsequent WIP PR I'm introducing another guard on whether the batcher will use blob transactions based on the SysConfig state (https://github.com/roberto-bayardo/optimism/pull/2).  I think this change will resolve this issue once I implement the TODO there to only activate blobs if the L1 block number from the sysconfig update event is beyond the blob activation time.

Reply:
Hmm I don't think that we have risk of a L2 reorg without waiting a # of confirmations after - the invariant must hold that `BlobsEnabledL1Timestamp >= L1 DecunTimestamp`. Could you explain in more depth the scenario you are thinking of?

We also want to not have this solely dependent on the timestamp, we want to have an override that can turn it on/off because for chains with very low throughput it will likely make it more expensive

Reply:
Issue is the batcher can't control exactly which L1 block a batcher transaction appears in regardless of using local timestamp or L1 derived timestamp. Consider this: batcher might submit a blob tx to the L1 txpool after blobsenabledL1 timestamp, but then the L1 might reorg to a point before the blobsenabledL1 timestamp, which can result in the blob tx to appear in an L1 block before the blobsenabledtimestamp.  The result here is that chain derivation would ignore that blob transaction.

Maybe this is a non-issue once I introduce the config option to turn batcher's use of blobs on/off. We can default it to "off" and only recommend turning it on well after the blobsEnabledTimestamp has passed.

Reply:
I'm going to resolve this conversation based on the call we had today. We'll implement a runtime "Strategy" flag that's up for the chain operator to set appropriately (e.g. it would never be set on a chain where Eclipse wasn't already active).